sdk/messaging_servicebus/Cargo.toml

```diff
@@ -15,13 +15,17 @@ categories = ["api-bindings"]
 edition = "2021"
 
 [dependencies]
-azure_core = { path = "../core", version = "0.2" }
+azure_core = "0.2"
```

Comment:
Why?

Reply:
I was using this crate in a local project to test it and, with `azure_core = { path = "../core", version = "0.2" }`, the compiler screamed at me that I was using two different `azure_core` versions (i.e., locally I had `azure_core = "0.2"`). After I changed it like so, it all worked 🤷

Reply:
This should be changed back. If you also want to use `azure_core` as a direct dependency locally, you should write `azure_core = { path = "../YOUR/AZURE_CORE/CHECKOUT", version = "0.2" }` .

Reply:
I don't want to use `azure_core` as a local dependency on my project, I want to use `azure_core = "0.2"`. With that said, why do I get errors for using two different `azure_core` versions if both `messaging_servicebus` and my local project use version `0.2`? That is, regardless of it being local or not...? 

For reference, this is the error I get:
```rs
error[E0308]: mismatched types
  --> crates/mq-azure-servicebus/src/lib.rs:33:13
   |
33 |             http_client,
   |             ^^^^^^^^^^^ expected trait `azure_core::http_client::HttpClient`, found trait `HttpClient`
   |
   = note: expected struct `Arc<(dyn azure_core::http_client::HttpClient + 'static)>`
              found struct `Arc<dyn HttpClient>`
   = note: perhaps two different versions of crate `azure_core` are being used?

For more information about this error, try `rustc --explain E0308
```

This is w/ my local project depending on `azure_core = "=0.2"`, and `messaging_servicebus` depending on `azure_core =  { path = "../core", version = "=0.2" }`. If I change `messaging_servicebus` to depend on `azure_core = "0.2"`, the problem is fixed. 

P.S. I'm getting the `HttpClient` the exact same way I do in the tests: `let http_client = azure_core::new_http_client();
`

Reply:
In any case~ done ✅

Reply:
The note `perhaps two different versions of crate `azure_core` are being used?` points to exactly what is happening.

You are using two versions of `azure_core` even though they are both labeled as version 0.2. One is the local copy (from the local `messaging_servicebus`) and one is from crates.io (from your direct dependency). The correct way to fix this is for your project to also depend on the local copy of `azure_core`.

Reply:
@rylev ~ I feel like that can't be the case. I, for example, use the `storage_blobs` crate just fine without having to depend on a local `azure_core`, and that crate itself depends on a local `azure_core` dependency...

Reply:
You're right~ I didn't know Rust would get confused by using two crates of equal versions from different registries. Thanks, Ryan!